articles/configuration/msbuild.md

```diff
@@ -1,3 +1,41 @@
+
+> [!NOTE]
+> ASP.NET Core projects implicitly include these files, so an ASP.NET Core project file will not need to import them.
+
+## TypeScript Tools Version
+Since a particular machine may have multiple versions of TypeScript installed, MSBuild allows you to set the TypeScriptToolsVersion property in your project file to specify which version to use.
+If this property is not set, MSBuild defaults to using the newest version of TypeScript that is installed. However, if you are using the TypeScript NuGet package instead, its version will override any setting in the project file.
+
+## Input Files
+There are two ways that files can be recognized by and therefore built with the project: either by the project file itself or by way of a tsconfig.json file.
+### With TSConfig
+One option for allowing MSBuild to recognize what TypeScript files are part of your project is by way of a tsconfig.json file. This configuration file can either be explicitly associated with your .csproj in the Content item list, or it can be simply included as part of the directory tree rooted at the directory containing your project file. Note that in the latter case, configuration files within a folder titled "node_modules", "bower_components", or "platforms" will not be considered in order to avoid duplicate symbol errors.
+
+The tsconfig.json can either explicitly enumerate the files intended to be compiled (in the "includes" property) or it can exclude files that should not be compiled (in the "excludes" property). The file also allows you to set compile options that will override those set in the TypeScript Build page of your project's Properties. For more details, refer to [the discussion of tsconfig.json files](xref:tsconfig).
+
+### Without TSConfig
+
+If there is no discoverable tsconfig.json, MSBuild relies on the project file to determine what files to include. For this scenario, TypeScript files must be included using the TypeScriptCompile item list.
+
+## Incremental Build
+
+The `Inputs` and `Outputs` attributes of certain targets, such as `CompileTypeScript` or `CompileTypeScriptWithTSConfig` allow MSBuild to determine whether or not to skip the target. For targets that have these attributes defined, MSBuild compares the timestamps of the files specified by `Inputs` with those specified by `Outputs`. Since there is not always a one-to-one mapping between input files and output files (e.g., if `--outFile` is specified), MSBuild will rebuild all output files if at least one of the input files has been updated.
```

Comment:
Correct me if I'm wrong but we don't have incremental build _yet_, so this might be confusing to people who are using current/older versions of TypeScript. I would recommend either removing this section (it's not thaaat important to document) or being specific about which version of TypeScript it applies to.

Reply:
added a comment about starting in TS 2.7 for now. I think it's important to have this documented in case something goes wrong and people need some way to find out why their build is getting skipped.